src/posthog-surveys.ts

```diff
@@ -27,6 +28,23 @@ export class PostHogSurveys {
         }
     }
 
+    getMatchingUrl(url?: string): boolean {
+        if (!url) return true
+
+        // If the url string starts and ends with / it is meant to be a regular expression
+        if (url.startsWith('/') && url.endsWith('/')) {
```

Comment:
not a big fan of this implicit matching, because if someone wanted to test for `/test/` , we'll end up matching `/test` as well. Not too big a deal, but indeed as you say there's an education component involved.

I think we can make this more explicit perhaps, have a toggle with the url field that mentions whether it's a regex or not, so users don't have to worry about adding custom `/` to the front and back, which will get removed (very confusing as the end user). And just input a clear regex when they select the isRegex checkbox.

So in posthog-js, that would mean you have a second parameter telling you whether it's a regex or not.

Reply:
@neilkakkar Thank you for the feedback! Would that turn the conditions type into `conditions: { url?: string; selector?: string; seenSurveyWaitPeriodInDays?: number, isRegex?: boolean } | null`? I wouldn't want to introduce any breaking changes on the url type, but having the isRegex boolean in the conditions object could make it seem like it also applies to the selector. On the other hand, since the checkbox would be controlled within posthog, that might be ok since the user would be explicitly applying it to the url. What do you think?

Reply:
I don't _love_ having to introduce another parameter for this especially because this is more like an add on than a requirement to using URLs with surveys, but if we do want to properly support url with regex, I agree it's better to go with a more complete solution. 

Instead of a checkbox though, we can make it more like our filters where the user has options to choose whether URL "matches regex" or "contains" the property value, similar to this:

<img width="620" alt="Screen Shot 2023-10-02 at 12 18 40 AM" src="https://github.com/PostHog/posthog-js/assets/25164963/3af0ef13-79ee-45c5-8fa3-55b5f8cc6a8a">

Except it would only be "matches regex" and "contains" for now 😅 

And the name of the parameter underneath the hood could be like `urlMatchType` where the match type can be "contains" or "regex" or "exact"

That way we can retain more flexibility than just having `isRegex` !

Reply:
It is an extra amount of work though, so feel free to let us know if it is, and the team will be happy to work on it together 😃

Reply:
Great call, makes more sense 👍

Reply:
I like the extra flexibility and matching an existing pattern!

Reply:
> It is an extra amount of work though, so feel free to let us know if it is, and the team will be happy to work on it together 😃

I think I can handle the changes in this code easily enough, but may need help on the posthog side. I have been on vacation, which is why I had such a delay replying to your comments. Is there a goal or timeline for having this change in place or hitting a release date? If so, I might not be able to get the posthog side done fast enough.

Reply:
It's fine to just complete posthog-js and I'm happy to take up the posthog side of things.

Alongside `conditions.url` I'll pass in a `conditions.urlMatchType` which currently will be one of `regex, exact, contains` which you can match by accordingly.